packages/core/test/unit/html/linkProcessor.test.js

```diff
@@ -35,6 +35,83 @@ const mockConfig = {
   ],
 };
 
+test('Test converting .md to .html', () => {
+  const mockLink = '<a href="/index.md">Test</a>';
+  const mockNode = cheerio.parseHTML(mockLink)[0];
+
+  const EXPECTED_RESULT = '<a href="/index.html">Test</a>';
+
+  linkProcessor.convertMdAndMbdExtToHtmlExt(mockNode);
+
+  expect(cheerio.html(mockNode)).toEqual(EXPECTED_RESULT);
+});
+
+test('Test converting .mbd to .html', () => {
+  const mockLink = '<a href="/index.mbd">Test</a>';
+  const mockNode = cheerio.parseHTML(mockLink)[0];
+
+  const EXPECTED_RESULT = '<a href="/index.html">Test</a>';
+
+  linkProcessor.convertMdAndMbdExtToHtmlExt(mockNode);
+
+  expect(cheerio.html(mockNode)).toEqual(EXPECTED_RESULT);
+});
+
+test('Test converting .md to .html with URL fragment', () => {
+  const mockLink = '<a href="/index.md#test-1">Test</a>';
+  const mockNode = cheerio.parseHTML(mockLink)[0];
+
+  const EXPECTED_RESULT = '<a href="/index.html#test-1">Test</a>';
+
+  linkProcessor.convertMdAndMbdExtToHtmlExt(mockNode);
+
+  expect(cheerio.html(mockNode)).toEqual(EXPECTED_RESULT);
+});
+
+test('Test converting .mbd to .html with URL fragment', () => {
+  const mockLink = '<a href="/index.mbd#test-1">Test</a>';
+  const mockNode = cheerio.parseHTML(mockLink)[0];
+
+  const EXPECTED_RESULT = '<a href="/index.html#test-1">Test</a>';
+
+  linkProcessor.convertMdAndMbdExtToHtmlExt(mockNode);
+
+  expect(cheerio.html(mockNode)).toEqual(EXPECTED_RESULT);
+});
+
+test('Test no conversion when no-convert attribute is present', () => {
+  const mockLink = '<a href="/index.md#test-1" no-convert>Test</a>';
+  const mockNode = cheerio.parseHTML(mockLink)[0];
+
+  const EXPECTED_RESULT = '<a href="/index.md#test-1" no-convert>Test</a>';
+
+  linkProcessor.convertMdAndMbdExtToHtmlExt(mockNode);
+
+  expect(cheerio.html(mockNode)).toEqual(EXPECTED_RESULT);
+});
+
+test('Test no conversion when neither .md nor .mbd extensions are present', () => {
+  const mockLink = '<a href="/index">Test</a>';
+  const mockNode = cheerio.parseHTML(mockLink)[0];
+
+  const EXPECTED_RESULT = '<a href="/index">Test</a>';
+
+  linkProcessor.convertMdAndMbdExtToHtmlExt(mockNode);
+
+  expect(cheerio.html(mockNode)).toEqual(EXPECTED_RESULT);
+});
+
+test('Test no conversion when both .md and .mbd extensions are present', () => {
```

Comment:
if I read this right, this can be generalised into a test for `.md/mbd` appearing in the middle

Reply:
Hmm not sure if I fully understand this part, but I've added new test cases which I think may cover this. For example, `// Test conversion when both .md and .mbd are present, and .mbd is extension, with URL fragment`. Let me know if I'm still missing out on this!